Add Docker and Compose setup

Noodlefu opened the pull request:
Introduces Dockerfile and docker-compose.yml to containerize MareSynchronos services, including main server, auth, services, static files, and dependencies (Postgres, Redis). Also adds example.env for environment variable configuration.

Noodlefu committed:
Add Docker and Compose setup

Introduces Dockerfile and docker-compose.yml to containerize MareSynchronos services, including main server, auth, services, static files, and dependencies (Postgres, Redis). Also adds example.env for environment variable configuration.

GiR-Zippo commented:
it's in the repo:
[Docker Setup ](https://github.com/Eauldane/SnowcloakServer/tree/main/Docker)

Noodlefu committed:
Add config change event and health check to service client

Introduces a ConfigChangedEvent and subscribes to configuration changes. Adds a health check for the configuration host before starting the update loop, and ensures proper disposal of the change subscription on shutdown.

Noodlefu committed:
Revert "Add config change event and health check to service client"

This reverts commit 275aa21a22b40e95c141d0ce7838c54aa7b6ea67.
